exam/models/exam.py

```diff
@@ -74,57 +82,148 @@ class ExamExam(models.Model):
 
     @api.constrains('start_date', 'end_date')
     def check_date_exam(self):
-        if self.end_date < self.start_date:
-            raise UserError(_('End date of Exam should be \
-                              greater than start date'))
-
+        '''Method to check constraint of exam start date and end date'''
+        for rec in self:
+            if rec.end_date < rec.start_date:
+                raise ValidationError(_('Exam end date should be \
+                                  greater than start date'))
+            for line in rec.exam_schedule_ids:
+                if line.timetable_id:
+                    for tt in line.timetable_id.exam_timetable_line_ids:
+                        if not rec.start_date <= tt.exm_date <= rec.end_date:
+                            raise ValidationError(_('Invalid Exam Schedule\
+                            \n\nExam Dates must be in between Start\
+                            date and End date !'))
+
+    @api.constrains('active')
+    def check_active(self):
+        '''if exam results is not in done state then raise an
+        validation Warning'''
+        result_obj = self.env['exam.result']
+        if not self.active:
+            for result in result_obj.search([('s_exam_ids', '=', self.id)]):
+                if result.state != 'done':
+                    raise ValidationError(_('Kindly,mark as done %s\
+                    examination results') % (self.name))
+
+    active = fields.Boolean('Active', default="True")
     name = fields.Char("Exam Name", required=True)
     exam_code = fields.Char('Exam Code', required=True, readonly=True,
                             default=lambda obj:
                             obj.env['ir.sequence'].next_by_code('exam.exam'))
-    standard_id = fields.Many2many('school.standard',
-                                   'school_standard_exam_rel', 'standard_id',
+    standard_id = fields.Many2many('standard.standard',
```

Comment:
Sir, This change is possible but standard_id field is used in many places inside our code. So it is critical to change the field name.